mongorestore/mongorestore_test.go

```diff
@@ -63,6 +68,59 @@ func getRestoreWithArgs(additionalArgs ...string) (*MongoRestore, error) {
 	return restore, nil
 }
 
+func BenchmarkCPUAndMemoryConsumption(b *testing.B) {
+	files, err := ioutil.ReadDir(benchmarkDataRootDir)
+
+	if err != nil {
+		b.Fatalf("Could not find test directory: %v\n", err)
+	}
+
+	for _, file := range files {
+		if file.Mode().IsRegular() {
+			testName := fmt.Sprintf("benchmark for %v", file.Name())
+
+			b.Run(testName, func(b *testing.B) {
+				// Setup arguments and MongoRestore context
+				benchFilePath := filepath.Join(benchmarkDataRootDir, file.Name())
+				collectionName := strings.TrimRight(file.Name(), filepath.Ext(file.Name()))
+
+				args := []string{
+					benchFilePath,
+					DBOption,
+					benchmarkDBName,
+					CollectionOption,
+					collectionName,
+				}
+
+				restore, err := getRestoreWithArgs(args...)
+
+				if err != nil {
+					b.Fatalf("Failed to generate restore context: %v", err)
+				}
+
+				defer func() {
+					client, err := restore.SessionProvider.GetSession()
+					defer restore.Close()
+
+					if err != nil {
```

Comment:
[nit] You shouldn't put code between a line that assigns an error and the line that checks that error.

Along with my suggestion above, I advise moving the `defer restore.Close()` line higher up, anyway. Perhaps after the error check on line 97, i.e.:
```
...
restore, err := getRestoreWithArgs(args...)
if err != nil { // notice there is _no_ newline between the line assigning err and the conditional checking it
    b.Fatalf("...")
}

defer restore.Close()
...
```